CRM/Pledge/Form/Payment.php

```diff
@@ -122,29 +135,24 @@ public function buildQuickForm() {
    * Process the form submission.
    */
   public function postProcess() {
-    // get the submitted form values.
     $formValues = $this->controller->exportValues($this->_name);
-    $params = array();
-    $formValues['scheduled_date'] = CRM_Utils_Date::processDate($formValues['scheduled_date']);
-    $params['scheduled_date'] = CRM_Utils_Date::format($formValues['scheduled_date']);
-    $params['currency'] = CRM_Utils_Array::value('currency', $formValues);
-    $now = date('Ymd');
+    $params = [
+      'id' => $this->_id,
+      'scheduled_date' => $formValues['scheduled_date'],
+      'currency' => $formValues['currency'],
+    ];
 
-    if (CRM_Utils_Date::overdue(CRM_Utils_Date::customFormat($params['scheduled_date'], '%Y%m%d'), $now)) {
```

Comment:
this formatting also stripped now

Reply:
We need to keep the second param otherwise it uses YmdHis as the comparison so the strings don't match and the comparison doesn't work correctly in CRM_Utils_Date::overdue